content/v2.0/reference/flux/stdlib/pushbullet/endpoint.md

```diff
@@ -20,7 +20,12 @@ import "pushbullet"
 
 pushbullet.endpoint(
   url: "https://api.pushbullet.com/v2/pushes",
-  token: ""
+  token: "",
+  mapFn: (r) => ({
```

Comment:
oh, I am sorry it's not like that:

```
e = p.endpoint(token)

d
  |> e(mapFn: (r) =>
  ({r with
    title: "hello from REPL",
    text: "Value: ${r._value}"}))()
```

Th endpoint is a function, that given a `mapFn` returns a transformation that can be used when you pipe in data 👍

Reply:
@affo I'm not sure I understand. This specific codeblock is just meant to show the function parameters and example values. The example below this shows data being piped in. Are the examples incorrect?

Reply:
Yep.

`pushbullet.endpoint()` does not have a `mapFn` parameter.

It returns a function that has a `mapFn` param, as I show in the example above.

That's actually the same for pagerduty/slack.

I think it should be ok to do the same as for them in docs 👍 

I see now it is kind of wrong also there:

 - https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/slack/endpoint/
 - https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/pagerduty/endpoint/

If you look at theFlux code for them (pgduty for example:

```
endpoint = (url=defaultURL) =>
    (mapFn) =>
        (tables=<-) => tables
            |> dedupKey()
            |> map(fn: (r) => {
                obj = mapFn(r: r)
                return {r with _sent: string(v: 2 == (sendEvent(pagerdutyURL: url,
                    routingKey: obj.routingKey,
                   .......
                ) / 100))}
            })
```

You can see that `endpoint` only takes `url`.

It then returns a function that accepts `mapFn`, that returns a transformation (or a function that accepts streams of tables).

Reply:
The comments above `endpoint` are pretty explanatory:

```
// `endpoint` creates the endpoint for the PushBullet external service.
// It will push notifications of type `note`.
// If you want to push something else, see `genericEndpoint`.
// `url` - string - URL of the PushBullet endpoint. Defaults to: "https://api.pushbullet.com/v2/pushes".
// `token` - string - token for the PushBullet endpoint.
// The returned factory function accepts a `mapFn` parameter.
// The `mapFn` must return an object with `title` and `text` fields as defined in the `pushNote` function arguments.
```

I think you could use those 🤔